StoreKitUnitTests/StoreTransactionTests.swift

```diff
@@ -0,0 +1,106 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  StoreTransactionTests.swift
+//
+//  Created by Nacho Soto on 1/10/22.
+
+import Nimble
+@testable import RevenueCat
+import StoreKitTest
+import XCTest
+
+class StoreTransactionTests: StoreKitConfigTestCase {
+
+    func testSK1DetailsWrapCorrectly() async throws {
+        try AvailabilityChecks.iOS15APIAvailableOrSkipTest()
+
+        let product = MockSK1Product(mockProductIdentifier: Self.productID)
+        let payment = SKPayment(product: product)
+
+        let sk1Transaction = MockTransaction()
+        sk1Transaction.mockPayment = payment
+        sk1Transaction.mockTransactionDate = Date()
+        sk1Transaction.mockTransactionIdentifier = UUID().uuidString
+
+        let transaction = StoreTransaction(sk1Transaction: sk1Transaction)
+
+        expect(transaction.sk1Transaction) === sk1Transaction
+
+        expect(transaction.productIdentifier) == Self.productID
+        expect(transaction.purchaseDate) === sk1Transaction.mockTransactionDate
+        expect(transaction.transactionIdentifier) == sk1Transaction.mockTransactionIdentifier
+    }
+
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    func testSK2DetailsWrapCorrectly() async throws {
+        try AvailabilityChecks.iOS15APIAvailableOrSkipTest()
+
+        let sk2Transaction = try await self.simulateAnyPurchase()
+
+        let transaction = StoreTransaction(sk2Transaction: sk2Transaction)
+
+        // Can't use `===` because `SK2Transaction` is a `struct`
```

Comment:
I opened https://github.com/Quick/Nimble/pull/955 so that `Nimble` doesn't even allow that.